### PR TITLE
Align Bazel on Windows with CMake's MSVC flags

### PR DESCRIPTION
**Pull request by rongjiecomputer**
Make Bazel on Windows build log less noisy.

### CODE REVIEW

**Comment by rongjiecomputer** [comment]
`-WHOLEARCHIVE:setargv.obj` is moved from `LINKOPTS` to `cc_library(name = "protoc_lib")` as it is the only place that needs this. Adding this flag in `LINKOPTS` will cause LNK4042 warning (object specified more than once; extras ignored).

`-WHOLEARCHIVE:` is changed to `-DEFAULTLIB:` as `-WHOLEARHIVE` can only be used for `.lib` library since VS2017 (LNK1356 error).

**Comment by rongjiecomputer** [comment]
If possible, I want this PR be included in 3.6.0 so that I can convince Tensorflow and gRPC to update dependency that contains this PR.

File: BUILD [hunk 1]
```diff
@@ -22,7 +22,21 @@ config_setting(
 MSVC_COPTS = [
     "/DHAVE_PTHREAD",
     "/wd4018", # -Wno-sign-compare
+    "/wd4065", # switch statement contains
```
Comment (xfxyjwf) [inline BUILD]: This one is duplicated.

**Comment by xfxyjwf** [comment]
I think this PR is too late to make into the 3.6.0 release.

File: BUILD (hunk 1 above)
Comment (rongjiecomputer) [reply BUILD]: Done.

**Comment by rongjiecomputer** [comment]
@xfxyjwf In that case, I will try to see if Tensorflow and gRPC are willing to pin to a commit near 3.6.0 after 3.6.0 is released. I thought this might still make it since 3.6.0 is still in rc2 and the changelog isn't updated yet #4713.

**Comment by rongjiecomputer** [comment]
@xfxyjwf Ping.

**Pull request merged**